DatadogLogs/Sources/RemoteLogger.swift

```diff
@@ -178,14 +189,21 @@ internal final class RemoteLogger: LoggerProtocol {
                     return
                 }
 
+                // Bit of a hack - add the `includeBinaryImages` attribute back in if it was in the original log
+                // TODO: RUM-4072 This double generates a backtrace - this should be optimized to use cached binary images instead
+                var rumAttributes: [String: Encodable] = combinedAttributes
+                if addBinaryImages {
+                    rumAttributes[CrossPlatformAttributes.includeBinaryImages] = true
+                }
+
                 self.core?.send(
                     message: .baggage(
                         key: ErrorMessage.key,
                         value: ErrorMessage(
                             message: log.error?.message ?? log.message,
                             type: log.error?.kind,
                             stack: log.error?.stack,
-                            attributes: .init(combinedAttributes)
+                            attributes: .init(rumAttributes)
                         )
                     )
                 )
```

Comment:
suggestion/ I think there is an easier way. We could add `binaryImages` to the `ErrorMessage` that is sent over the bus. Note that [`BinaryImage` conforms](https://github.com/DataDog/dd-sdk-ios/blob/b63af60df8c6417866b5acd70689d8622962641c/DatadogInternal/Sources/Models/CrashReporting/BinaryImage.swift#L10) to [`PassthroughAnyCodable`](https://github.com/DataDog/dd-sdk-ios/blob/b63af60df8c6417866b5acd70689d8622962641c/DatadogInternal/Sources/Codable/AnyEncoder.swift#L664-L673) meaning that it is transmitted without actual coding (no perf hit).

Upon [receiving the new `ErrorMessage` in RUM](https://github.com/DataDog/dd-sdk-ios/blob/b63af60df8c6417866b5acd70689d8622962641c/DatadogRUM/Sources/Integrations/ErrorMessageReceiver.swift#L41) it is passed to internal [`Monitor.addError()`](https://github.com/DataDog/dd-sdk-ios/blob/b63af60df8c6417866b5acd70689d8622962641c/DatadogRUM/Sources/RUMMonitor/Monitor.swift#L533) which emits `RUMAddCurrentViewErrorCommand`. This command has a [broad initializer that accepts `binaryImages`](https://github.com/DataDog/dd-sdk-ios/blob/b63af60df8c6417866b5acd70689d8622962641c/DatadogRUM/Sources/RUMMonitor/RUMCommand.swift#L179-L191), so we should be fine 🔗 👍.

Reply:
So I remembered the reason I didn't do this, though it's potentially trivial to fix. `ErrorMessageReceiver` uses the [public `Monitor.addError` method](https://github.com/DataDog/dd-sdk-ios/blob/258d9ff2e144f393587e031d2a2719e6826b82c7/DatadogRUM/Sources/Integrations/ErrorMessageReceiver.swift#L41) to add the error to the monitor, which we obviously do not want to modify to accept binary images.

The answer is fairly straightforward, but adds a new method to the internal interface on Monitor just for this.